asf_installer_survey/notebooks/section_6_analysis.py

```diff
@@ -0,0 +1,1670 @@
+
+# Save to .csv and display
+df_617_crosstab_owner_V2.to_csv("../../outputs/section6/csv/recoded/617_owner_V2_recoded.csv")
+df_617_crosstab_owner_V2
+
+# %%
+# Generate figure
+groupedbar(df_617_crosstab_owner_V2,
+           "617_owner_V2",
+           "RQ 6.17 How do the reasons that survey participants give for only offering, or only intending to offer, repairs, servicing and maintenance to heat pump systems they’ve installed (but not for all other systems) differ depending on the size of the company they work for or own?")
+
+# %%
+# Generate figure
+donut(df_617_crosstab_owner_V2,
+           "617_owner_V2",
+           "RQ 6.17 How do the reasons that survey participants give for only offering, or only intending to offer, repairs, servicing and maintenance to heat pump systems they’ve installed (but not for all other systems) differ depending on the size of the company they work for or own?")
+
+# %%
+# CompanySizeEmployee V2
+# Display dataframes to be merged
+df_606_crosstab_employee_V2
+df_615_crosstab_employee_V2
+
+# %%
+# CompanySizeEmployee V2
+# Create new df_615_crosstab_crosstab_employee_V2 with dummy column to match df_606_crosstab_employee_V2
+df_615_crosstab_employee_V2_mapped = df_615_crosstab_employee_V2.copy()
+df_615_crosstab_employee_V2_mapped["Only work with certain heat pump models"] = 0
+
+# Merge dataframes
+# N.B. Need to rename header of columns - SQ44 should be RQ6.17
+df_617_crosstab_employee_V2 = df_606_crosstab_employee_V2.add(df_615_crosstab_employee_V2_mapped)
+
+# Save to .csv and display
+df_617_crosstab_employee_V2.to_csv("../../outputs/section6/csv/recoded/617_employee_V2_recoded.csv")
+df_617_crosstab_employee_V2
+
+# %%
+# Generate figure
+groupedbar(df_617_crosstab_employee_V2,
+           "617_employee_V2",
+           "RQ 6.17 How do the reasons that survey participants give for only offering, or only intending to offer, repairs, servicing and maintenance to heat pump systems they’ve installed (but not for all other systems) differ depending on the size of the company they work for or own?")
+
+# %%
+# Generate figure
+donut(df_617_crosstab_employee_V2,
+           "617_employee_V2",
+           "RQ 6.17 How do the reasons that survey participants give for only offering, or only intending to offer, repairs, servicing and maintenance to heat pump systems they’ve installed (but not for all other systems) differ depending on the size of the company they work for or own?")
+
+# %% [markdown]
+# ### Q 6.18 | Why haven’t survey participants that intend on offering heat pump repairs, servicing and maintenance not done so thus far?
+# - SQ: 50
+# - Sample:
+# - Standard_output
+# <p>
+# - N.B. Employees don’t answer this question AND this is a Select all that apply type question
+# <p>
+# Update: Free text responses recoded on 2024-04-18
+
+# %%
+# Print possible answers and check counts
+data[col.q50[0]].value_counts()
+
+# %%
+def transform_50(x):
```

Comment:
More general comment, but a lot of the functions are a very similar structure and we are rewriting them over and over again, could we not create a package with functions such as transform and then just import them like we have done for e.g. `crosstable`.

Reply:
Thanks Aidan! Have taken the time now to write a general function for "transform" and added it to the "data_setup.py" file that is imported at the start.